File: classes/LLMs/examples/pdf_classification/code/python/process_pdfs.py
from langchain.document_loaders import PyPDFLoader
from langchain.text_splitter import RecursiveCharacterTextSplitter
import os

def process_pdfs(folder_path):
    processed_texts = []
    
    # Process each PDF in the folder
    for filename in os.listdir(folder_path):
        if filename.endswith('.pdf'):
            file_path = os.path.join(folder_path, filename)
            
            # Load and process PDF
            loader = PyPDFLoader(file_path)
            pages = loader.load()
            
            # Split text into manageable chunks
            text_splitter = RecursiveCharacterTextSplitter(
                chunk_size=1000,
                chunk_overlap=200
            )
            texts = text_splitter.split_documents(pages)
            
            # Store processed text with filename
            processed_texts.append({
                'filename': filename,
                'content': texts
            })
            
    return processed_texts

File: classes/LLMs/examples/pdf_classification/code/R/process_pdfs.R

# Function to process PDFs
process_pdfs <- function(path_to_pdfs) {
  # Get list of PDF files
  pdf_files <- list.files(
    path = path_to_pdfs,
    pattern = "\\.pdf$",
    full.names = TRUE
  )
  
  # Process each PDF
  processed_papers <- map_df(pdf_files, function(file) {
    # Extract text from PDF
    text <- pdf_text(file) %>%
      paste(collapse = " ") %>%
      # Clean text
      str_replace_all("\r\n", " ") %>%
      str_replace_all("\\s+", " ") %>%
      str_trim()
    
    # Split into chunks of approximately 1000 characters
    chunks <- text %>%
      str_split_1(boundary("word")) %>%
      split(ceiling(seq_along(.)/200)) %>%
      map_chr(paste, collapse = " ")
    
    # Create dataframe
    tibble(
      filename = basename(file),
      chunk_id = seq_along(chunks),
      content = chunks
    )
  })
  
  return(processed_papers)
}